pkg/virt-controller/services/template.go

```diff
@@ -939,10 +944,21 @@ func (t *templateService) renderLaunchManifest(vmi *v1.VirtualMachineInstance, t
 		compute.Env = append(compute.Env, k8sv1.EnvVar{Name: varName, Value: resourceName})
 	}
 
-	if _, ok := vmi.Labels[debugLogs]; ok {
+	virtLauncherLogVerbosity := t.clusterConfig.GetVirtLauncherVerbosity()
+
+	if verbosity, isSet := vmi.Labels[logVerbosity]; isSet || virtLauncherLogVerbosity != virtconfig.DefaultVirtLauncherLogVerbosity {
+		// Override the cluster wide verbosity level if a specific value has been provided for this VMI
+		verbosityStr := fmt.Sprint(virtLauncherLogVerbosity)
+		if isSet {
+			verbosityStr = verbosity
+		}
+		compute.Env = append(compute.Env, k8sv1.EnvVar{Name: ENV_VAR_VIRT_LAUNCHER_LOG_VERBOSITY, Value: verbosityStr})
+	}
+
+	if _, ok := vmi.Labels[debugLogs]; ok || virtLauncherLogVerbosity > EXT_LOG_VERBOSITY_THRESHOLD {
```

Comment:
I wonder if we should consider using the downward api here. If annotations on the launcher pod can propagate down to to the compute container as env vars, then maybe we could dynamically turn on/off debugging for live vmis after their pod is up and running.

Reply:
>Also, it seems that specific label in fieldPath was added only in 1.19 [1]
How can we go around this if we still support 1.17 and 1.18?

oh, bummer. yeah the only other option would be to use a volume and mount all the annotations. I'm not 100% sure whether or not the annotation content mounted as files in the pod gets dynamically updated when the annotations change on the pod.  

```
volumes:
    - name: pod-annotations
      downwardAPI:
        items:
          - path: "annotations"
            fieldRef:
              fieldPath: metadata.annotations
```


I don't think we have to pursue this dynamic verbosity on the launcher pod right away. I do think the downward api might give us a path to doing this in the future though.